src/App.tsx

```diff
@@ -1,98 +1,76 @@
 import React from 'react';
 import './App.css';
-import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+export class App extends React.Component {
+  state = {
+    itemsPerPageOptions: [3, 5, 10, 20],
+    perPage: 5,
+    currentPage: 1,
+    total: 42,
+  };
 
-export const App: React.FC = () => {
-  return (
-    <div className="container">
-      <h1>Items with Pagination</h1>
+  setItemsPerPage = (num:number) => {
+    this.setState({
+      perPage: num,
+      currentPage: 1,
+    });
+  };
 
-      <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
-      </p>
+  handlePageChange = (page:number) => {
+    this.setState({
+      currentPage: page,
+    });
+  };
 
-      <div className="form-group row">
-        <div className="col-3 col-sm-2 col-xl-1">
-          <select
-            data-cy="perPageSelector"
-            id="perPageSelector"
-            className="form-control"
-          >
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
-          </select>
-        </div>
+  render() {
+    const {
+      perPage,
+      itemsPerPageOptions,
+      currentPage,
+      total,
+    } = this.state;
 
-        <label htmlFor="perPageSelector" className="col-form-label col">
-          items per page
-        </label>
-      </div>
+    return (
+      <div className="container">
+        <h1>Items with Pagination</h1>
+
+        <p className="lead" data-cy="info">
+          {`Page ${currentPage} (items ${(perPage * (currentPage - 1)) + 1} - ${currentPage * perPage > total ? total : currentPage * perPage} of ${total})`}
```

Comment:
Consider breaking the long string template into multiple lines for better readability.
Also, it would be great to create variables for some values